tests/testthat/test-derive_vars_dtm.R

```diff
@@ -203,3 +203,31 @@ test_that("No re-derivation is done if --DTF variable already exists", {
   expect_equal(expected_output, actual_output)
 
 })
+
+test_that("Issue a warning if --DTM already exists", {
+  expected_output <- tibble::tribble(
+    ~XXSTDTC, ~ASTDTM, ~ASTDTF, ~ASTTMF,
+    "2019-07-18T15:25:40", ymd_hms("2019-07-18T15:25:40"), NA_character_, NA_character_,
+    "2019-07-18T15:25", ymd_hms("2019-07-18T15:25:00"), NA_character_, "S",
+    "2019-07-18T15", ymd_hms("2019-07-18T15:00:00"), NA_character_, "M",
+    "2019-07-18", ymd_hms("2019-07-18T00:00:00"), NA_character_, "H",
+    "2019-02", ymd_hms("2019-02-15T00:00:00"), "D", "H",
+    "2019", ymd_hms("2019-06-15T00:00:00"), "M", "H",
+    "2019---07", ymd_hms("2019-06-15T00:00:00"), "M", "H"
+  ) %>%
+    mutate(ASTDTM = as_iso_dtm(ASTDTM))
+
+  actual_output <- derive_vars_dtm(
+    input,
+    new_vars_prefix = "AST",
+    dtc = XXSTDTC,
+    date_imputation = "MID"
+  )
+
+
+  expect_equal(
+    expected_output,
+    actual_output
+  )
+
+})
```

Comment:
This should be a test that makes sure the `max_dates` work as expected.

Reply:
maybe we need to call max_dates and min_dates in such test

Reply:
Yes, exactly.

Reply:
Th is the truncated result comparing with the content of test_that() message as dicussed yesterday, but I guess no impact on test result as we get the expected warning

![image](https://user-images.githubusercontent.com/7486109/144053937-c5dcf379-c108-4bc0-9e00-34212a8f75f4.png)